sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -187,15 +188,98 @@ bool DistortionCorrector<T>::isInputValid(sensor_msgs::msg::PointCloud2 & pointc
   return true;
 }
 
+template <class T>
+bool DistortionCorrector<T>::azimuthConversionExists(sensor_msgs::msg::PointCloud2 & pointcloud)
+{
+  if (!isInputValid(pointcloud)) return false;
+
+  sensor_msgs::PointCloud2Iterator<float> it_x(pointcloud, "x");
+  sensor_msgs::PointCloud2Iterator<float> it_y(pointcloud, "y");
+  sensor_msgs::PointCloud2Iterator<float> it_azimuth(pointcloud, "azimuth");
+
+  auto next_it_x = it_x;
+  auto next_it_y = it_y;
+  auto next_it_azimuth = it_azimuth;
+
+  if (it_x != it_x.end() && it_x + 1 != it_x.end()) {
+    next_it_x = it_x + 1;
+    next_it_y = it_y + 1;
+    next_it_azimuth = it_azimuth + 1;
+  } else {
+    RCLCPP_WARN(
+      node_->get_logger(),
+      "Current point cloud only has a single point. Could not calculate the formula.");
+    return false;
+  }
+
+  for (; next_it_x != it_x.end();
+       ++it_x, ++it_y, ++it_azimuth, ++next_it_x, ++next_it_y, ++next_it_azimuth) {
+    auto current_cartesian_rad = autoware::universe_utils::opencv_fastAtan2(*it_y, *it_x);
+    auto next_cartesian_rad = autoware::universe_utils::opencv_fastAtan2(*next_it_y, *next_it_x);
+
+    // If the angle exceeds 180 degrees, it may cross the 0-degree axis,
+    // which could disrupt the calculation of the formula.
+    if (
+      abs(*next_it_azimuth - *it_azimuth) >= autoware::universe_utils::pi ||
+      abs(next_cartesian_rad - current_cartesian_rad) >= autoware::universe_utils::pi) {
+      RCLCPP_WARN(
+        node_->get_logger(),
+        "Angle between two points exceeds 180 degrees. Iterate to next point ...");
+      continue;
+    }
```

Comment:
There should be four explicit cases (both azimuth and cartesian_rad less than PI, bigger/equal to PI, or the first or latter is less and the other is bigger/equal).

Handling these cases explicitly (should only have effects on the sign of `b` and sign/offset of `a`) would get rid of the necessity of looping until the 'easy' case is found.